Fetch and redeem provider keys using tokens

jp1ac4 opened the pull request:
This PR implements #1578 and #1579.

Please note the following:
- I've left the cosigner option enabled for initial testing as part of this PR, but will disable it afterwards (#1584 ).
- Using cosigner keys is currently possible in all templates, including the Simple Inheritance template, but given that there is only one key per path in this template, it would not be possible to continue with a cosigner key selected. I could disable the option to select cosigner keys in this template in a follow-up PR, but in any case there will be a warning so the user will be aware of the problem.
- Safety net paths can only be added in the Custom template for now.
- ~~This cannot yet be tested with Liana Connect.~~ Can be tested on Signet with Liana Connect.
- ~~The URL for fetching & redeeming keys is still localhost and I will update it once the service is ready.~~
- ~~The GUI layout needs to be refined as part of this PR.~~ No further GUI layout changes planned.

Please also note that I've unhelpfully squashed everything into a single commit. I hope to be able to split these out into smaller commits, but wanted to check the overall approach first and there may also be some conflicts with other PRs.

jp1ac4 commented:
Since creating this PR, I've made the following additions/changes:
- added space between buttons "+ Add recovery option" and "+ Add Safety Net"
- check for duplicate alias after fetching key (so user must change alias if two cosigner keys from same provider are used, for example)
- default timelock for standard recovery path will correspond to 1 year instead of MAX blocks - 1
- tooltip added to "+ Add Safety Net" button
- if error occurs redeeming tokens after installation is complete, warning is added to installer log and installer log will not be deleted
- token form value is now trimmed
- check when fetching key that if existing key has same fingerprint, it must have same provider key kind

edouardparis commented:
When the fetched xpub has wrong information: like wrong network or kind, I think it would be better to mark the form::Value as invalid and have the form invalid message instead of an error on top of the modal.

jp1ac4 commented:
I've changed it so that if the fetched key fails any sanity checks, the form will be invalid and the corresponding warning will be shown below in red, instead of using the error card at the top of the modal.

I've also changed how provider keys are listed in the edit key modal to include the key kind and token. This same display will also be used when we've just fetched the key via the token but not yet saved it by clicking the Apply button. In this case, the button containing the provider key info will not be clickable as there's not yet a saved key to select.

jp1ac4 commented:
Updated to include provider UUID and name in the key settings.
